src/java/org/jivesoftware/openfire/commands/admin/user/SetUserAdditionalProperty.java

```diff
@@ -0,0 +1,184 @@
+package org.jivesoftware.openfire.commands.admin.user;
+
+import org.dom4j.Element;
+import org.jivesoftware.openfire.XMPPServer;
+import org.jivesoftware.openfire.commands.AdHocCommand;
+import org.jivesoftware.openfire.commands.SessionData;
+import org.jivesoftware.openfire.user.UserNotFoundException;
+import org.jivesoftware.openfire.user.UserWrapper;
+import org.xmpp.forms.DataForm;
+import org.xmpp.forms.FormField;
+import org.xmpp.packet.JID;
+
+import java.util.Collections;
+import java.util.List;
+import java.util.Map;
+
+/**
+ *  An ad-hoc command to retrieve the additional properties of the user.
+ */
+public class SetUserAdditionalProperty extends AdHocCommand {
+
+    private XMPPServer xmppServer;
+    private UserWrapper propertyStore;
+
+    @Override
+	public String getCode() {
+        return "http://jabber.org/protocol/admin#set-user-additional-property";
+    }
+
+    @Override
+	public String getDefaultLabel() {
+        return "Set an additional user property";
+    }
+
+    @Override
+	public int getMaxStages(SessionData data) {
+        return 1;
+    }
+
+    @Override
+	public void execute(SessionData sessionData, Element command) {
+
+        Map<String, List<String>> data = sessionData.getData();
+
+        Element note = command.addElement("note");
+
+        String accountJid;
+        String propertyKey;
+        String propertyValue;
+
+        try {
+            accountJid = data.get("accountjid").get(0);
+            propertyKey = data.get("key").get(0);

```

Comment:
Check for empty JID (to avoid NPE) and key (to avoid setting the empty property key).

Add tests to confirm empty key can't be set.